client-device-connection-infinispan/pom.xml

```diff
@@ -59,13 +61,23 @@
        </exclusion>
       </exclusions>
     </dependency>
+
+    <!-- TODO: these are repeated from client-device-connection-base. Keep or remove? -->
```

Comment:
keep, if code in this module requires these to compile

Reply:
Do you know of a good way to get maven to discover whether you have explicit dependency declarations on stuff that you reference directly from your module code?

I'm going through this by trial and error, removing all dependencies and then adding them one by one with a wildcard exclusion on all transitive dependencies, in order to find all direct dependencies.

Reply:
You shouldn't bother too much. If you run into a case where you immediately see that it should/could be replaced with a transitive dep then do so. Otherwise, keep it as is ...

Reply:
I've done my best to clean up here without causing to much chaos.